### PR TITLE
cross-referenced Array type in docs

### PR DESCRIPTION
**Pull request by rawwar**
Updated Array explanation to contain a reference to Array class definition.

image

### CODE REVIEW

File: docs/source/reference/types.rst [hunk 1]
```diff
@@ -76,7 +76,7 @@ complex128              c16              double-precision complex number
 Arrays
 ------
 
-The easy way to declare array types is to subscript an elementary type
+The easy way to declare :class:`~numba.types.Array` types is to subscript an elementary type
```
Comment (stuartarchibald) [inline docs/source/reference/types.rst:79]: Please line wrap to 80 characters, thanks.

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, looks good, just needs line wrapping.

**Comment by stuartarchibald** [comment]
Rendered build is here: https://numba--6843.org.readthedocs.build/en/6843/reference/types.html#arrays

**Comment by rawwar** [comment]
I updated the docs.

File: docs/source/reference/types.rst (hunk 1 above)
Comment (stuartarchibald) [reply docs/source/reference/types.rst:79]: It looks like https://github.com/numba/numba/pull/6843/commits/8452bdefa986fa4113f53196b32cf53764319e01 changed the line below and not this line?

Comment (rawwar) [reply docs/source/reference/types.rst]: Yes, I am sorry. I will correct this.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch.

**Pull request merged**